src/Features/ExternalAccess/OmniSharp/Options/OmniSharpSolutionAnalyzerConfigOptionsUpdater.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.Options;
+using Roslyn.Utilities;
+using Microsoft.CodeAnalysis.ImplementType;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.Options;
+
+using Workspace = CodeAnalysis.Workspace;
+
+internal static class OmniSharpSolutionAnalyzerConfigOptionsUpdater
+{
+    internal static void UpdateOptions(Workspace workspace, OmniSharpEditorConfigOptions editorConfigOptions)
+    {
+        try
+        {
+            workspace.SetCurrentSolution(UpdateOptions, changeKind: WorkspaceChangeKind.SolutionChanged);
```

Comment:
oh.  this is... concerning.  @tmat  for thoughts on someone actively calling into SetCurrentSolution. 

Is tehre a reason they need this?  do they not have their own Workspace suclass they can call through to do this?

Reply:
Looking at `Workspace.TryApplyChanges` and it appears to copy the `FallbackAnalyzerOptions`. Perhaps I can just switch to using that instead.